app/views/track.js

```diff
@@ -118,6 +127,45 @@ const TrackView = Mn.LayoutView.extend({
         e.stopPropagation();
     },
 
+    setClass() {
+        this.className = ''
+        this.order = [];
+
+        // Check if track is in upNext
+        let tracks = application.upNext.get('tracks');
+        tracks.each((track, index) => {
+            if (track.get('_id') == this.model.get('_id')) {
+                this.className += ' playlist-upNext';
+                this.order.push({ id: 'upNext', order: index });
+                return;
```

Comment:
I think your `return` here is useless, It's always the end of the function. What you should do is an early return, something like:

```js
tracks.each((track, index) => {
    if (track.get('_id') != this.model.get('_id')) { return }
    this.className += ' playlist-upNext';
    this.order.push({ id: 'upNext', order: index });
```

but I still think a `where` (or a `findWhere`) instruction is more relevant as long you try to find an occurence of the current (aka only one) model in a given collection.